File: backend/templates/typescript/src/StressTest.ts
import { HttpClientFactory } from './http/HttpClientFactory.js';
import { HttpClientFactoryImpl } from './http/HttpClientFactoryImpl.js';

export abstract class StressTest {
  private httpFactory = new HttpClientFactoryImpl();

  public interval(): number {
    return 1000;
  }
  public abstract test(userId: string): void;

  public get http(): HttpClientFactory {
    return this.httpFactory;
  }
}

File: backend/templates/python/src/stress_test.py
from abc import ABC, abstractmethod
from web.client_factory import HttpClientFactory


class StressTest(ABC):
    def __init__(self):
        self._http_factory = HttpClientFactory()

    def interval(self) -> int:
        return 1000

    @abstractmethod
    def test(self, user_id: str) -> None:
        pass

    @property
    def http(self) -> HttpClientFactory:
        return self._http_factory
